Sources/ApodiniDeploy/RemoteHandlerInvocationManager.swift

```diff
@@ -0,0 +1,308 @@
+        case .remotely(let targetNode):
+            guard let runtime = RHIIE.deploymentProviderRuntime else {
+                return eventLoop.makeFailedFuture(ApodiniDeployError(message: "Unable to find runtime"))
+            }
+            // The targetEndpoint is on a different node, dispatch the invocation there
+            
+            // TODO if theres a guarantee that an EndpointParameter's id is the same as the id of the @Parameter it was generated from,
+            // we can combine these two sets into one
+            var alreadyProcessedParamKeyPaths: Set<AnyKeyPath> = []
+            var alreadyProcessedEndpointParamIds: Set<UUID> = []
+
+            let invocationParams: [HandlerInvocation<H>.Parameter] = collectedInputParams.map { collectedParam in
+                // The @Parameter property wrapper declaration in the handler
+                let handlerParamId = (targetEndpoint.handler[keyPath: collectedParam.handlerKeyPath] as! AnyParameterID).value
+                let endpointParam = targetEndpoint.findParameter(for: handlerParamId)!
+                if !alreadyProcessedParamKeyPaths.insert(collectedParam.handlerKeyPath).inserted {
+                    print("Parameter '\(endpointParam.name)' specified multiple times in remote handler invocation")
+                }
+                if !alreadyProcessedEndpointParamIds.insert(endpointParam.id).inserted {
+                    print("Endpoint parameter with id '\(endpointParam.id)' matched multiple times in remote handler invocation")
+                }
+                return HandlerInvocation<H>.Parameter(
+                    stableIdentity: endpointParam.lk_stableIdentity,
+                    name: endpointParam.name,
+                    value: collectedParam.value as! HandlerInvocation<H>.Parameter.Value
+                )
+            }
+            let missingParamNames: [String] = targetEndpoint.parameters
+                .filter { !$0.hasDefaultValue && !alreadyProcessedEndpointParamIds.contains($0.id) }
+                .map(\.name)
+            guard missingParamNames.isEmpty else {
+                fatalError("Missing parameters in remote handler invocation: \(missingParamNames)")
+            }
+            
+            assert(handlerId == targetEndpoint.identifier)
+            
+            do {
+                let handlerInvocation = HandlerInvocation<H>(handlerId: handlerId, targetNode: targetNode, parameters: invocationParams)
+                let runtimeHandlingResult = try runtime.handleRemoteHandlerInvocation(handlerInvocation)
+                
+                switch runtimeHandlingResult {
+                case .result(let future):
+                    return future
+                case .invokeDefault(let url):
+                    let requestUrl = url
+                        .appendingPathComponent("__apodini")
+                        .appendingPathComponent("invoke")
```

Comment:
Yes, i had that thought as well. I decided to underscore the first path component, because this is an internal API and i didn't want it to cause any name clashes with user-defined handlers, but some other parts of Apodini (eg the grpc exporter) already define additional endpoints which are not underscored (eg `/apodini/proto`). These probably should be unified in some way. Generally speaking I'd say that all additional endpoints (ie endpoints which are not generated from a `Handler`) should be under one common "internal" path. A notable exception to this would be the OpenAPI definition which is currently served at `/openapi`. I don't know if that is required by the standard or could also be moved into the internal group.

Reply:
As far as I know there is not standard path that an OpenAPI specification has to be served on. I would propose to move every internal functionality behind a reserved `/apodini/` route. So `/apodini/openapi`, `/apodini/swaggerui`, `/apodini/proto`, and `/apodini/invoke` would all be located at these routes.
I have documented this in an issue #248